### ISSUE TITLE
Support passing existing connections

### ISSUE THREAD

**Issue by idomic** [ploomber/jupysql#16]
The initialization locks the context to a specific engine. [See example](https://github.com/catherinedevlin/ipython-sql/issues/216)

We'd like to allow users to connect to an engine directly by reference (similar to SQL alchemy).

**Comment by idomic** [comment]
Another [great example](https://github.com/catherinedevlin/ipython-sql/issues/199) of this use case (this was solved via the Duckdb team)

**Comment by edublancas** [comment]
I'm thinking something like this:

```python
from sqlalchemy import create_engine

engine = create_engine("...")
```

Then:

```python
%sql $engine
```

IPython supports variable expansion via `$var`, but I'm unsure if this will pass the actual object or only its string representation.

**Comment by edublancas** [comment]
It's a bit unclear to me the original issue in the ipython-sql repo. I don't understand if the user wants to re-use an existing engine (what I described in the comment above). Or they want to re-use an engine they created with `%sql` for later queries. Maybe we can ask him and bring the conversation here?

**Comment by idomic** [comment]
I think that's the first option.
@machow thoughts?

**Comment by idomic** [comment]
I think [this one](https://github.com/catherinedevlin/ipython-sql/issues/201) will also be solved once an engine can be passed dynamically?  @zzFluke

**Comment by Alex-Monahan** [comment]
This is still a nice and helpful feature, even if DuckDB can already read dataframes without it! One use case is if you store your dataframes in a dict, you would want to be able to register them with DuckDB (since it won't find them djrectly anymore).

**Comment by idomic** [comment]
@Alex-Monahan thanks for the feedback! Do you think it's related to this issue or a completely new one?

**Comment by Alex-Monahan** [comment]
The ability to either pass in an engine or refer to the engine variable and call a function directly on the connection would work in this case. I think that is what this proposes, but I am not certain!

**Comment by idomic** [comment]
Awesome, yes it is!

**Comment by edublancas** [comment]
hey @Alex-Monahan, @machow, this has been fixed!

Available in version 0.5.1. [Docs here.](https://jupysql.readthedocs.io/en/latest/connecting.html#using-an-existing-sqlalchemy-engine-engine)

Let us know if you have any issues/feedback!

**Comment by idomic** [comment]
fyi: @pfmoore @gwerbin @rehno-lindeque @ellisonbg @ckald
